### PR TITLE
Add table to keep track of schedule data migrations 1/5

### PR DESCRIPTION
**Pull request by prha**
## Summary
Adds a schema migration to add a table to keep track of schedule storage migrations.

If the storages are colocated in the same DB, this is a no-op

## Test Plan
BK, added backcompat tests

### CODE REVIEW

**Comment by prha** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7182** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7182" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: python_modules/dagster/dagster/core/storage/schedules/sqlite/sqlite_schedule_storage.py [hunk 1]
```diff
@@ -43,22 +43,29 @@ def config_type(cls):
...
             if not (db_revision and head_revision):
                 ScheduleStorageSqlMetadata.create_all(engine)
                 engine.execute("PRAGMA journal_mode=WAL;")
                 stamp_alembic_rev(alembic_config, connection)
+                should_migrate_data = True
+
+        schedule_storage = cls(conn_string, inst_data)
+        if should_migrate_data:
+            schedule_storage.migrate()
+            schedule_storage.optimize()
```
Comment (gibsondan) [inline python_modules/dagster/dagster/core/storage/schedules/sqlite/sqlite_schedule_storage.py:66]: is there any risk of things failing partway through and leaving the schema created but the data migrations not run?

**Review by gibsondan — APPROVED** [review]
feedback there would all apply to the run storag emigrations as well, so not blocking. the fn name in particular is worth considering though I think

File: python_modules/dagster/dagster/core/storage/schedules/base.py [hunk 2]
```diff
@@ -123,5 +123,11 @@ def get_tick_stats(self, origin_id: str):
     def upgrade(self):
         """Perform any needed migrations"""
 
+    def migrate(self, print_fn: Optional[Callable] = None, force_rebuild_all: bool = False):
+        """Call this method to run any required data migrations"""
+
+    def optimize(self, print_fn: Optional[Callable] = None, force_rebuild_all: bool = False):
+        """Call this method to run any optional data migrations for optimized reads"""
```
[the remark this answers is not in the document]
Comment (prha) [reply python_modules/dagster/dagster/core/storage/schedules/base.py:130]: Yeah, I think all of this stuff needs to be unified and renamed.  For one, if the secondary index table is shared across storages (they are colocated), then the migration names have to share a namespace.  I took a crack at doing a whole refactor of this but it ended up being too big and decided to fork that off into a different thing.

**Pull request merged**